websites/B/Battlefield Portal/presence.ts

```diff
@@ -0,0 +1,141 @@
+const presence = new Presence({
+  clientId: "919182644296683520"
+});
+async function ParseGrpc(
+  sessionId: string,
+  playgroundId: string
+): Promise<string> {
+  const a = await fetch(
+    "https://kingston-prod-wgw-envoy.ops.dice.se/web.communitygames.CommunityGames/getPlayground",
+    {
+      credentials: "omit",
+      headers: {
+        "User-Agent":
+          "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:95.0) Gecko/20100101 Firefox/95.0",
+        Accept: "*/*",
+        "Accept-Language": "en-US,en;q=0.5",
+        "x-gateway-session-id": sessionId,
+        "x-dice-tenancy": "prod_default-prod_default-kingston-common",
+        "Content-Type": "application/grpc-web+proto",
+        "X-User-Agent": "grpc-web-javascript/0.1",
+        "X-Grpc-Web": "1",
+        "Sec-Fetch-Dest": "empty",
+        "Sec-Fetch-Mode": "cors",
+        "Sec-Fetch-Site": "cross-site"
+      },
+      referrer: "https://portal.battlefield.com/",
+      body: `\u0000\u0000\u0000\u0000&\u0012$${playgroundId}`,
+      method: "POST",
+      mode: "cors"
+    }
+  );
+  if (!a.ok) throw new Error("Failed");
+
+  return a.text();
+}
+
+let called = false,
+  playgroundId: string = null,
+  info: string[] = null;
+
+const EditingStamp = Math.floor(Date.now() / 1000),
+  presenceData: PresenceData = {
+    largeImageKey: "img1"
+  };
+
+presence.on("UpdateData", async () => {
+  const [, sessionId] = document.cookie.match(
+      new RegExp("(^| )sessionId=([^;]+)")
+    ),
+    url = document.URL;
+
+  if (url.search("=") > 0) playgroundId = url.split("=").pop();
+
+  if (sessionId && playgroundId) {
+    if (document.readyState === "complete") {
+      if (!called) {
+        ParseGrpc(sessionId, playgroundId).then((t) => {
+          info = t
+            .split("$")[1]
+            .split("AiMaxCount_PerTeam")[0]
+            .split("\u000c")[1]
+            .split("*");
+          if (!info) {
+            called = false;
+            presenceData.details = null;
+          } else {
+            presence.info(info[0]);
+            presence.info(info[1]);
+          }
+        });
+        called = true;
+      }
+    } else called = false;
+  }
+  if (info) {
+    presence
+      .getSetting("name")
+      .then((t) =>
+        t
+          ? (presenceData.details = `Making ${info[0]}`)
+          : (presenceData.details = "")
```

Comment:
```suggestion
        if (t) presenceData.details = `Making ${info[0]}`;
```

Reply:
If I do this then presence won't be updated . As persenceData is defined outside outside the event function

Reply:
Sure then